android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/protocol/MobileWalletAdapterServer.java

```diff
@@ -256,6 +256,7 @@ private void onAuthorizationComplete(@NonNull NotifyOnCompleteFuture<Authorizati
                 }
                 o.put(ProtocolContract.RESULT_ACCOUNTS, accounts);
                 o.put(ProtocolContract.RESULT_WALLET_URI_BASE, result.walletUriBase); // OK if null
+                o.put(ProtocolContract.RESULT_WALLET_ICON, result.walletIcon);
```

Comment:
We should have a check (somewhere, not necessarily here) that the scheme of this URI is `data:`

Reply:
I added 2 checks: 1 in the `LocalScenario` constructor to check the uri from the icon provider, and another in the `AuthorizationResult` constructor to check the icon uri parameter